crates/ruff_python_formatter/src/other/arguments.rs

```diff
@@ -223,7 +223,7 @@ fn is_huggable_string_argument(
     arguments: &Arguments,
     context: &PyFormatContext,
 ) -> bool {
-    if string.is_implicit_concatenated() || !string.is_multiline(context.source()) {
+    if string.is_implicit_concatenated() || !string.is_multiline(context) {
         return false;
     }
```

Comment:
I'm not sure I follow here. Are you saying that both the code snippet should result in the formatting where the f-string is on the same line as the call expression (`call(f"{`)?

Reply:
This is not directly related to your PR, but I noticed it when reviewing it because you changed `is_multiline`. We have to make a decision on the idiomatic formatting for "hugging multiline strings" for f-strings. 

I'm leaning towards that we should only "hug" if the f-string itself contains any multiline string literal, but not if any inner-expression is multiline. Similar to prettier: 

```js
call(
  `${
    aaaaaaaaaaaaaaaaaaaaaaaaaaaa +
    `test more
    aaa` +
    morrrrrrrrrrrrrrrrrrrrrr
  }`,
);
```

So what I'm saying is that we should probably not hug for the both cases above.

Reply:
I suggest that we add an explicit `is_triple_quoted` check here. I don't think we ever want this to apply to any non-triple quoted strings. And let's add some test cases that cover f-strings too (cases where we don't want the layout to apply as well as cases where it should apply)